src/main/java/seedu/address/logic/parser/ViewAvailableCommandParser.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.ViewAvailableCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.OnDateTimeApptPredicate;
+import seedu.address.model.timeslots.OnDateTimeSlotPredicate;
+import seedu.address.model.timeslots.Timeslots;
+
+/**
+ * Parser class for ViewAvailableCommand
+ */
+public class ViewAvailableCommandParser implements Parser<ViewAvailableCommand> {
+    /**
+     * Parses the input from the user and returns the ViewAvailableCommand
+     * @param args String input from user
+     * @return An instance of ViewAvailableCommand
+     * @throws ParseException when format of ViewAvailableCommand is wrongly used
+     */
+    public ViewAvailableCommand parse(String args) throws ParseException {
```

Comment:
For some reason 31-02-2024 passed in results in Feb 29, 2024 search.
Can you find some way to prevent invalid dates inputs?